src/Commands/Commenting.cs

```diff
@@ -35,28 +32,55 @@ private static CommandProgression Execute(Action<DocumentView> action)
 
         private static void Comment(DocumentView doc)
         {
-            SnapshotSpan spans = doc.TextView.Selection.SelectedSpans.First();
-            Collection<ITextViewLine> lines = doc.TextView.TextViewLines.GetTextViewLinesIntersectingSpan(spans);
-
-            foreach (ITextViewLine line in lines.Reverse())
+            var snapshot = doc.TextBuffer.CurrentSnapshot;
+            int start = doc.TextView.Selection.Start.Position.Position;
+            int end = doc.TextView.Selection.End.Position.Position;
+            var editsession = doc.TextBuffer.CreateEdit();
+            while (start < end)
             {
-                doc.TextBuffer.Insert(line.Start.Position, Constants.CommentChars[0]);
+                var line = snapshot.GetLineFromPosition(start);
+                editsession.Insert(line.Start.Position, Constants.CommentChars[0] + " ");
+                start = line.EndIncludingLineBreak.Position;
             }
+            editsession.Apply();
```

Comment:
The `editsession` must always be disposed after use. It can be done by wrapping `using` around it. Same for the `Uncomment` method

Reply:
Done